activiti-cloud-audit-service/activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/ProcessDeletedEventConverter.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Order;
+import org.springframework.data.jpa.domain.Specification;
+
+public class ProcessDeletedEventConverter extends BaseEventToEntityConverter {
+
+    private final String PROCESS_INSTANCE_ID = "processInstanceId";
+
+    private final String TIMESTAMP = "timestamp";
+
+    private final String MISSING_PROCESS_INSTANCE = "Process Instance %s not found";
+
+    private final String INVALID_PROCESS_INSTANCE_STATE = "Process Instance %s is not in a valid state: %s";
+
+    private final Set<String> VALID_EVENT_TYPES =
+        Arrays.asList(CloudRuntimeEventType.PROCESS_CANCELLED, CloudRuntimeEventType.PROCESS_COMPLETED).stream().map(Enum::name)
+            .collect(Collectors.toSet());
+
+    private final EventsRepository eventsRepository;
+
+    public ProcessDeletedEventConverter(EventsRepository eventsRepository, EventContextInfoAppender eventContextInfoAppender) {
+        super(eventContextInfoAppender);
+        this.eventsRepository = eventsRepository;
+    }
+
+    @Override
+    public String getSupportedEvent() {
+        return ExtendedCloudProcessRuntimeEvents.PROCESS_DELETED.name();
+    }
+
+    @Override
+    protected ProcessDeletedAuditEventEntity createEventEntity(CloudRuntimeEvent cloudRuntimeEvent) {
+        ProcessAuditEventEntity event = findEvent(cloudRuntimeEvent.getProcessInstanceId());
+        return new ProcessDeletedAuditEventEntity(event, (CloudProcessDeletedEvent) cloudRuntimeEvent);
+    }
+
+    @Override
+    protected CloudRuntimeEventImpl<?, ?> createAPIEvent(AuditEventEntity auditEventEntity) {
+        ProcessDeletedAuditEventEntity processDeletedAuditEventEntity = (ProcessDeletedAuditEventEntity) auditEventEntity;
+
+        return new CloudProcessDeletedEventImpl(processDeletedAuditEventEntity.getEventId(),
+                                                processDeletedAuditEventEntity.getTimestamp(),
+                                                processDeletedAuditEventEntity.getProcessInstance());
+    }
+
+    protected ProcessAuditEventEntity findEvent(String processInstanceId) {
+        Specification<AuditEventEntity> specification = new EventSpecificationsBuilder().with(new SpecSearchCriteria(PROCESS_INSTANCE_ID, SearchOperation.EQUALITY, processInstanceId)).build();
+        List<AuditEventEntity> events = eventsRepository.findAll(specification, Sort.by(Order.desc(TIMESTAMP)));
+        AuditEventEntity lastEvent = events.stream().findFirst().orElseThrow(() -> new IllegalStateException(String.format(MISSING_PROCESS_INSTANCE, processInstanceId)));
+        if(VALID_EVENT_TYPES.contains(lastEvent.getEventType())) {
+            return ProcessAuditEventEntity.class.cast(lastEvent);
+        } else {
+            throw new IllegalStateException(String.format(INVALID_PROCESS_INSTANCE_STATE, processInstanceId, lastEvent.getEventType()));
+        }
+    }
```

Comment:
We can remove it, but this type of events can not be effectively validated inside RB since it does not store any information about the completed or cancelled processes, which means that we could end up for storing multiple PROCESS_DELETED events in audit service.

Reply:
Done